jni/include/faiss_wrapper.h

```diff
@@ -44,7 +44,8 @@ namespace knn_jni {
         //
         // Return an array of KNNQueryResults
         jobjectArray QueryIndex_WithFilter(knn_jni::JNIUtilInterface * jniUtil, JNIEnv * env, jlong indexPointerJ,
-                                                                jfloatArray queryVectorJ, jint kJ, jintArray filterIdsJ);
+                                                                jfloatArray queryVectorJ, jint kJ, jlongArray filterIdsJ,
+                                                                jint filterIdsTypeJ);
```

Comment:
also instead of passing filterIdsJ and filterIdsTypeJ can we pass an object and then use JNI methods to get the array and the type. 

Thought-process: The number of parameters for this function is increasing and at some point in time the number of arguments will become un-manageable.

Reply:
i try to use this method as i create a new object `FilterIdsSelector`. but it would do many class `FindMethod` and need many unit test for class find safety. 
how abut make these change into a new pr for optimize the code readability

Reply:
Sure, we can tackle this in another PR.